scRNA-seq-advanced/setup/SCPCP000005-samples/Snakefile

```diff
@@ -0,0 +1,24 @@
+# Workflow for preprocessing SCPCP000005 data used for integration instruction
+# Raw files are expected to be present in the `config[raw_dir]` directory
+
+configfile: "config.yaml"
+workdir: config["base_dir"]
+
+rule target:
+  input:
+    expand(os.path.join(config['base_dir'], "{library}.rds"),
+           library = config["libraries"])
+
+rule prepare_libraries:
+  output:
+    expand(os.path.join(config['base_dir'], "{library}.rds"),
+           library = config["libraries"])
```

Comment:
For best practice all snakemake rules should work on a single sample at a time. I know this requires rewriting the notebook and removes the need for `purrr`, but it allows more parallelism! We also want to be sure to specify the inputs so we get an error if needed, but using wildcards to the parallelism works as expected.

We also want to take advantage of the log functionality. (If you want to use a notebook, the log could be the notebook output maybe?)

If you are using the workdir as listed, this can be a simple as
```suggestion
rule prepare_library:
  input: 
    filtered_rds = "raw/{library}_filtered.rds",
    celltype_tsv = "annotations/celltypes.tsv"
  output:
    "processed/{library}.rds"
  log: 
    "logs/{library_prep}.log"
```

I also had a later thought about whether to use the `workdir` directive at all, or if we should do the work in the setup directory. If we don't use workdir, then the input and output directories have to be a bit more extensive, but we can put the logs in the setup directory, which is kind of nice. This is the approach I took in https://github.com/AlexsLemonade/training-modules/pull/581, but whatever we decide, we should be consistent.